src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -371,151 +371,160 @@ internal unsafe void Start()
                 if (context is null)
                 {
                     // Flow suppressed, just run here
-                    ProcessEvents(numEvents.ToInt32(), eventPaths, eventFlags, eventIds, watcher);
+                    ProcessEvents(numEvents.ToInt32(), eventPaths, new Span<FSEventStreamEventFlags>(eventFlags, numEvents.ToInt32()), new Span<FSEventStreamEventId>(eventIds, numEvents.ToInt32()), watcher);
```

Comment:
Do we still need to pass in numEvents.ToInt32() as the first argument if both spans being passed in are of that length?

Reply:
Unfortunately, I can't create span of byte* which is why the first argument is passed as byte** with length as an extra parameter.

I was about to parse the string from byte** to ReadOnlySpan<char>[] however it would lead to allocate an array for all paths. If I do this it would open an opportunity to pass an FSEvent[] with consolidated information data instead of three arrays.

FSEvent would look like `{ ReadOnlySpan<char> path, FSEventStreamEventFlags flags, FSEventStreamEventId id }`.

Reply:
I didn't find any simple solution for now